qemu/tests/qemu_guest_agent.py

```diff
@@ -1291,6 +1291,115 @@ def gagent_check_get_disks(self, test, params, env):
                     test.fail("Disk %s alias is defferent "
                               "between guest and qga." % diskname)
 
+    @error_context.context_aware
+    def gagent_check_ssh_public_key_injection(self, test, params, env):
+        """
+        Execute commands related inject ssh_public_keys to guest agent,
+        and all of these following commands need to be checked via another
+        family command "guest-ssh-get-authorized-keys".
+        Steps:
+        1) Check basic function whether can work well.
+        2) check the result of adding new ssh keys that
+         one of them was existed.
+        3) remove ssh keys
+        4) reset ssh keys.
+        :param test: kvm test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment.
+        """
+
+        def ssh_key_test(operation, *keys, **kwargs):
+            guest_user = params['guest_user']
+            op_func = getattr(self.gagent, 'ssh_%s_authorized_keys' % operation)
+            op_func(guest_user, *keys, **kwargs)
+            keys_ga = self.gagent.ssh_get_authorized_keys("fedora")
+            session.cmd(params["add_line_at_end"] % guest_home_path)
+            keys_guest = session.cmd_output(cmd_guest_keys).strip()
+            _value_compared_ga_guest(keys_ga, keys_guest, operation)
+            return keys_ga, keys_guest
+
+        def _prepared_n_restore_env(status):
+            if status != "1":
+                session.cmd(params["cmd_add_user_set_passwd"])
+                session.cmd(params["set_setenforce"] % int(status))
+            else:
+                session.cmd(params["cmd_remove_user"])
+
+        def _generate_host_keys():
+            process.system(params["cmd_clean_keys"], shell=True)
+            status = process.system(params["ssh_keygen_cmd"],
+                                    shell=True, timeout=3)
+            if status:
+                test.error("Can not generate ssh key with no "
+                           "interaction, please have a check.")
+            cmd_get_hostkey = params["cmd_get_hostkey"]
+            host_key = process.system_output(cmd_get_hostkey,
+                                             shell=True, timeout=3)
+            host_key = host_key.decode(encoding="utf-8",
+                                       errors="strict")
+            return host_key
+
+        def _login_guest_test(guest_ip):
+            cmd_login_guest = params["test_login_guest"] % guest_ip
+            login_user = process.system_output(cmd_login_guest,
+                                               shell=True, timeout=5)
+            login_user = login_user.strip().decode(encoding="utf-8",
+                                                   errors="strict")
```

Comment:
```suggestion
            login_user = process.getoutput(cmd_login_guest,
                                           shell=True, timeout=5).strip()
```

Reply:
@6-dehan Please check the above comment, thanks.

Reply:
Idk exactly why it always get fails when I used 'process.getoutput', but 'process.system_output' can be passed.